src/SwedbankPay.Episerver.Checkout.OrderManagement/Steps/AuthorizePaymentStepBase.cs

```diff
@@ -5,24 +5,34 @@
 
 using SwedbankPay.Episerver.Checkout.Common;
 
+using System.Threading.Tasks;
+
 namespace SwedbankPay.Episerver.Checkout.OrderManagement.Steps
 {
     public abstract class AuthorizePaymentStepBase : PaymentStep
     {
-        public AuthorizePaymentStepBase(IPayment payment, IMarket market, ISwedbankPayClientFactory swedbankPayClientFactory) : base(payment, market, swedbankPayClientFactory)
+        protected AuthorizePaymentStepBase(IPayment payment, IMarket market, ISwedbankPayClientFactory swedbankPayClientFactory) : base(payment, market, swedbankPayClientFactory)
         {
         }
 
-        public override bool Process(IPayment payment, IOrderForm orderForm, IOrderGroup orderGroup, IShipment shipment, ref string message)
+        public override async Task<PaymentStepResult> Process(IPayment payment, IOrderForm orderForm, IOrderGroup orderGroup, IShipment shipment)
         {
             if (payment.TransactionType != TransactionType.Authorization.ToString())
             {
-                return Successor != null && Successor.Process(payment, orderForm, orderGroup, shipment, ref message);
+                var paymentStepResult = new PaymentStepResult();
+                if (Successor != null)
+                {
+                    paymentStepResult = await Successor.Process(payment, orderForm, orderGroup, shipment).ConfigureAwait(false);
+                    paymentStepResult.Status = paymentStepResult.Status;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove or correct this useless self-assignment.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-episerver-checkout/pullRequest?prid=5328029)